src/traces/parcoords/calc.js

```diff
@@ -0,0 +1,49 @@
+/**
+* Copyright 2012-2017, Plotly, Inc.
+* All rights reserved.
+*
+* This source code is licensed under the MIT license found in the
+* LICENSE file in the root directory of this source tree.
+*/
+
+'use strict';
+
+var hasColorscale = require('../../components/colorscale/has_colorscale');
+var calcColorscale = require('../../components/colorscale/calc');
+var Lib = require('../../lib');
+
+
+module.exports = function calc(gd, trace) {
+    var inputDimensions = trace.dimensions,
+        dimensions = [];
+    for(var i = 0; i < inputDimensions.length; i++) {
```

Comment:
Done, thank you @etpinard for working it out! Any way to preserve your authorship of this commit? Should my next feature branch e.g. Sankey simply be in the `plotly` repo? (I initially had no auth but now I probably do).  Minor note, I [needed to return a bit more](https://github.com/monfera/plotly.js/blob/9511e694dec465aa5ad483745f6dfec771ac98d6/src/traces/parcoords/calc.js#L28-L32) from `calc.js` b/c 3 unit tests failed  - I could have just returned `[trace]` but didn't like the idea of passing around an object with lots of properties when only some of them are needed; though I wondered if it would be a better fit for Plotly).